packages/jest-core/src/watch.ts

```diff
@@ -51,7 +55,11 @@ const INTERNAL_PLUGINS = [
   QuitPlugin,
 ];
 
-const RESERVED_KEY_PLUGINS = new Map([
+const RESERVED_KEY_PLUGINS = new Map<
+  // TODO: Should be WatchPlugin
```

Comment:
why does it not work?

Reply:
`Value of type 'typeof UpdateSnapshotsPlugin' has no properties in common with type 'WatchPlugin'. Did you mean to call it?`

Reply:
Oh, then this shouldn't actually be `WatchPlugin` (an instance) but something like `WatchPluginClass`

Reply:
`WatchPlugin` is an interface

Reply:
Yes, specifying it here would mean the map keys are instances implementing the interface, but we actually want constructors for instances of the interface.

Reply:
They all `extend BaseWatchPlugin` which implements the interface. Changing the key to `typeof BaseWatchPlugin` though leads to another error further down

![image](https://user-images.githubusercontent.com/1404810/53686922-fda85180-3d2d-11e9-9632-db64236233db.png)

Reply:
it's probably the `plugin.constructor` part that's the issue - `constructor` is typed as `Function` instead of giving us the `class`. Any other way of getting the class from an instance of it?

Reply:
https://github.com/jeysal/jest/commit/2824dab02cb5541d910543c08692ac5e6a31836c is the cleanest can come up with, at least it decouples from `BaseWatchPlugin` to "a constructor for something that implements WatchPlugin". But still requires `.constructor as` cast because TS just defines `Object.constructor` as `Function`

Reply:
thanks!